<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MultiRD • MultiRD</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MultiRD">
<meta property="og:description" content="MultiRD">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiRD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MultiRD.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">References</a>
</li>
<li>
  <a href="../articles/web/data.html">Data</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/chencxxy28/MultiRD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MultiRD_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MultiRD</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/chencxxy28/MultiRD/blob/master/vignettes/MultiRD.Rmd" class="external-link"><code>vignettes/MultiRD.Rmd</code></a></small>
      <div class="hidden name"><code>MultiRD.Rmd</code></div>

    </div>

    
    
<p><strong>MultiRD</strong> unifies multiple deconvolution schemes to infer cell type proportions from the target bulk RNA-seq data. Three unique features are embraced in this algorithm: first, <strong>MultiRD</strong> is able to incorporate extra biological information from external data sources enables (e.g., scRNA_seq and other bulk RNA_seq data from independent studies); second, <strong>MultiRD</strong> calibrates the reference-free algorithm by taking into account the proportion estimates from a reference-based approach; third, <strong>MultiRD</strong> is robust to incorrect information from any of the provided data sources.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"chencxxy28/MultiRD"</span><span class="op">)</span></code></pre></div>
<p>In this tutorial, we use SCDC (<a href="https://academic.oup.com/bib/article/22/1/416/5699815" class="external-link">Dong et al. (2021)</a>) for illustration to conduct reference-based deconvolution. The package can be installed by the code</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"meichendong/SCDC"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/chencxxy28/MultiRD" class="external-link">MultiRD</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/meichendong/SCDC" class="external-link">SCDC</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="data-input" class="section level2">
<h2 class="hasAnchor">
<a href="#data-input" class="anchor"></a>Data Input</h2>
<p><strong>MultiRD</strong> deconvolutes raw read counts data which is in <code>ExpressionSet</code> objects based on the package <a href="https://bioconductor.org/packages/release/bioc/html/Biobase.html" class="external-link"><em>Biobase</em></a>. As an illustration of the <strong>MultiRD</strong> framework, we conduct both simulation studies and real data analysis based on pancreatic islet data from <a href="https://www.pnas.org/content/111/38/13924" class="external-link">Fadista et al. (2014)</a> as presented in our paper. Three public scRNA-seq datasets from <a href="https://www.sciencedirect.com/science/article/pii/S2405471216302666?via%3Dihub" class="external-link">Baron et al. (2016)</a>, <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe et al. (2016)</a> and <a href="https://www.sciencedirect.com/science/article/pii/S155041311630434X?via%3Dihub" class="external-link">Xin et al. (2016)</a> are used. For simulations, we have a pre-specified <a href="https://github.com/chencxxy28/MultiRD/blob/master/vignettes/data/cmarkers.rds" class="external-link">correct</a> and <a href="https://github.com/chencxxy28/MultiRD/blob/master/vignettes/data/wmarkers.rds" class="external-link">incorrect</a> marker genes; for real data application, we have an expression dataset from <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005694" class="external-link">Bunt et al. (2015)</a> and pre-selected list of <a href="https://github.com/chencxxy28/MultiRD/blob/master/vignettes/data/fad_list_marker.rds" class="external-link">marker genes</a>. All processed data are available at the <a href="https://chencxxy28.github.io/MultiRD/articles/web/data.html">data download</a>.</p>
</div>
<div id="simulation-studies" class="section level2">
<h2 class="hasAnchor">
<a href="#simulation-studies" class="anchor"></a>Simulation Studies</h2>
<p>First, read in scRNA-seq data from our website</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">readRDSFromWeb</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ref</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/gzcon.html" class="external-link">gzcon</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="va">ref</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">seger</span> <span class="op">&lt;-</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/segerstolpe.rds"</span><span class="op">)</span>
<span class="va">baron</span> <span class="op">&lt;-</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/baron.rds"</span><span class="op">)</span>
<span class="va">xin</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/Xin_nonD.rds"</span><span class="op">)</span></code></pre></div>
<p>Second, use function <code><a href="../reference/generateBulk.html">generateBulk()</a></code> to create the <em>pseudo bulk</em> samples, where <code>ct.varname</code> specifies the name of cell type clustering result variable. <code>sample</code> specifies the name of subjects information variable. <code>ct.sub</code> specifies the names of cell types used to construct <em>pseudo bulk</em> samples. Here we provide an example where the <em>pseudo bulk</em> samples are generated from <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe et al. (2016)</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span>
<span class="va">pseudo.seger</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/generateBulk.html">generateBulk</a></span><span class="op">(</span><span class="va">seger</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>, sample <span class="op">=</span> <span class="st">"sample"</span>, ct.sub <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span>, nbulk <span class="op">=</span> <span class="fl">50</span>, low_s <span class="op">=</span> <span class="fl">0.3</span>, upp_s <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></code></pre></div>
<p>The generated <em>pseudo bulk</em> object contains a matrix of true cell type proportions (<code>pseudo.seger$truep</code>) and the <code>ExpressionSet</code> object (<code>pseudo.seger$pseudo_eset</code>).</p>
<p>MultiRD algorithm takes as input the target bulk data, subject-level cell type proportions from reference-based approach, and population-level cell type proportions. For illustration, we use the function <code><a href="https://rdrr.io/pkg/SCDC/man/SCDC_prop.html" class="external-link">SCDC_prop()</a></code> from SCDC package (<a href="https://academic.oup.com/bib/article/22/1/416/5699815" class="external-link">Dong et al. (2021)</a>). We regard the estimated cell type proportions from <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe et al. (2016)</a> as the correct subject-level cell type information (<code>c_refbase_est</code>) and the estimated ones with subject-label shuffle from <a href="https://www.sciencedirect.com/science/article/pii/S155041311630434X?via%3Dihub" class="external-link">Xin et al. (2016)</a> as the incorrect information (<code>w_refbase_est</code>).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span>
<span class="co">#correct subject-level proportions</span>
<span class="va">correct_refest</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/SCDC_prop.html" class="external-link">SCDC_prop</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="va">pseudo.seger</span><span class="op">[[</span><span class="st">"pseudo_eset"</span><span class="op">]</span><span class="op">]</span>, sc.eset <span class="op">=</span> <span class="va">seger</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>, sample <span class="op">=</span> <span class="st">"sample"</span>, weight.basis <span class="op">=</span> <span class="cn">T</span>, ct.sub <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span><span class="op">)</span>
<span class="va">c_refbase_est</span> <span class="op">=</span> <span class="va">correct_refest</span><span class="op">$</span><span class="va">prop.est.mvw</span>

<span class="co">#incorrect subject-level proportions</span>
<span class="va">incorrect_refest</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/SCDC_prop.html" class="external-link">SCDC_prop</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="va">pseudo.seger</span><span class="op">[[</span><span class="st">"pseudo_eset"</span><span class="op">]</span><span class="op">]</span>, sc.eset <span class="op">=</span> <span class="va">xin</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>,sample <span class="op">=</span> <span class="st">"sample"</span>, weight.basis <span class="op">=</span> <span class="cn">T</span>, ct.sub <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span><span class="op">)</span>
<span class="va">w_refbase_est</span> <span class="op">=</span> <span class="va">incorrect_refest</span><span class="op">$</span><span class="va">prop.est.mvw</span>
<span class="va">w_refbase_est</span> <span class="op">=</span> <span class="va">w_refbase_est</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">w_refbase_est</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span></code></pre></div>
<p>We provide the function <code>evaluate</code> to assess the performance of estimated proportions versus the true proportions based on mean absolute deviance (MAD, the smaller the better), concordance correlation coefficient (CCC, the larger the better), and Pearson correlation coefficient (Pearson, the larger the better).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">c_refbase_est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span>
<span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">w_refbase_est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></code></pre></div>
<p>We can obtain population-level mean proportions, with the one using <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe et al. (2016)</a> as correct (<code>c_ave_est</code>) and the one using <a href="https://www.sciencedirect.com/science/article/pii/S155041311630434X?via%3Dihub" class="external-link">Xin et al. (2016)</a> as incorrect (<code>w_ave_est</code>). The function <code><a href="../reference/pop.ct.prop.subj.html">pop.ct.prop.subj()</a></code> is used to extract such information given subject-level proportions. We also provide the true mean proportions (<code>true_ave</code>) from the target pseudo bulk samples. Besides, MultiRD also requires a list of prespecified marker genes, which could be selected from external scRNA-seq data based on pre-defined expression threshold, prior biological knowledge, or information reported in literature. The list of correct markers (<code>cmarkers.rds</code>) and the list of wrong markers (<code>wmarkers.rds</code>) for this simulation study are available in files <code>cmarkers.rds</code> and <code>wmarkers.rds</code> at <a href="https://chencxxy28.github.io/MultiRD/articles/web/data.html">data download</a>, respectively.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">c_list_marker</span> <span class="op">=</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/cmarkers.rds"</span><span class="op">)</span> <span class="co">#may write a function for selecting marker genes</span>
<span class="va">w_list_marker</span> <span class="op">=</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/wmarkers.rds"</span><span class="op">)</span> <span class="co">#may write a function for selecting marker genes</span>
<span class="va">c_ave_est</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.subj.html">pop.ct.prop.subj</a></span><span class="op">(</span><span class="va">c_refbase_est</span><span class="op">)</span>
<span class="va">w_ave_est</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.subj.html">pop.ct.prop.subj</a></span><span class="op">(</span><span class="va">w_refbase_est</span><span class="op">)</span>
<span class="va">true_ave</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.subj.html">pop.ct.prop.subj</a></span><span class="op">(</span><span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">true_ave</span></code></pre></div>
<pre><code>##      alpha       beta      delta      gamma 
## 0.62117169 0.20960298 0.08329970 0.08592562</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">c_ave_est</span></code></pre></div>
<pre><code>##      alpha       beta      delta      gamma 
## 0.60617872 0.23603323 0.08822580 0.06956226</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">w_ave_est</span></code></pre></div>
<pre><code>##       alpha        beta       delta       gamma 
## 0.893160143 0.020098938 0.007843283 0.078897635</code></pre>
<p>Finally, we integrate all information into MultiRD algorithm to conduct deconvolution for the <em>pseudo bulk</em> data. For illustration, we focus on the case with wrong marker genes (W), wrong subject-level proportions (W), and correct population-level mean proportions (R). The function <code><a href="../reference/MultiRD.onegroup.html">MultiRD.onegroup()</a></code> will return a list containing estimated cell type proportions corresponding to each tuning value (<code>est.prop</code>) and a sequence of goodness-of-fit values corresponding to each tuning value (<code>metrics</code>). The smaller the better.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MultiRD.object</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/MultiRD.onegroup.html">MultiRD.onegroup</a></span><span class="op">(</span>bulk.data<span class="op">=</span><span class="va">pseudo.seger</span><span class="op">[[</span><span class="st">"pseudo_eset"</span><span class="op">]</span><span class="op">]</span>,list.marker<span class="op">=</span><span class="va">w_list_marker</span>,celltype.unique<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span>,subject.level.proportion<span class="op">=</span><span class="va">w_refbase_est</span>,population.level.proportion<span class="op">=</span><span class="va">c_ave_est</span><span class="op">)</span></code></pre></div>
<p>The function <code>MultiRD.predict.props()</code> helps extract the recovered cell type proportions via MultiRD; the function <code><a href="../reference/reffree.predict.prop.html">reffree.predict.prop()</a></code> helps extract the estimated proportions via reference-free approach. We use the function <code>evaluate</code> to assess the performance of estimated proportions versus the true proportions based on MAD, CCC, and Pearson. We also provide the visualization via a dot plot using MAD and CCC. The following results show that our method (named MultiRD) has smaller MAD and higher CCC and Pearson than other methods even when misleading information is used.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MultiRD.est</span> <span class="op">=</span> <span class="fu"><a href="../reference/MultiRD.predict.prop.html">MultiRD.predict.prop</a></span><span class="op">(</span>MultiRD.output<span class="op">=</span><span class="va">MultiRD.object</span><span class="op">)</span>
<span class="va">reffree.est</span> <span class="op">=</span> <span class="fu"><a href="../reference/reffree.predict.prop.html">reffree.predict.prop</a></span><span class="op">(</span>MultiRD.output<span class="op">=</span><span class="va">MultiRD.object</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">MultiRD.est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></code></pre></div>
<pre><code>##      all.mad   all.ccc   all.cor
## 1 0.05077932 0.2265825 0.6944258</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">reffree.est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></code></pre></div>
<pre><code>##     all.mad   all.ccc   all.cor
## 1 0.1871329 0.0578328 0.6236459</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">w_refbase_est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></code></pre></div>
<pre><code>##     all.mad    all.ccc   all.cor
## 1 0.1413312 0.02907685 0.1222001</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Methods<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MultiRD"</span>,<span class="st">"Ref-free"</span>,<span class="st">"SCDC"</span>,<span class="st">"MultiRD"</span>,<span class="st">"Ref-free"</span>,<span class="st">"SCDC"</span>,<span class="st">"MultiRD"</span>,<span class="st">"Ref-free"</span>,<span class="st">"SCDC"</span>,<span class="st">"MultiRD"</span>,<span class="st">"Ref-free"</span>,<span class="st">"SCDC"</span><span class="op">)</span>, Cell_Types<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Alpha"</span>,<span class="st">"Alpha"</span>,<span class="st">"Alpha"</span>,<span class="st">"Beta"</span>,<span class="st">"Beta"</span>,<span class="st">"Beta"</span>, <span class="st">"Delta"</span>,<span class="st">"Delta"</span>,<span class="st">"Delta"</span>,<span class="st">"Gamma"</span>,<span class="st">"Gamma"</span>,<span class="st">"Gamma"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">MultiRD_results</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">MultiRD.est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">cell.type.eva</span>
<span class="va">reffree_results</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">reffree.est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">cell.type.eva</span>
<span class="va">SCDC_results</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">w_refbase_est</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">cell.type.eva</span>
<span class="va">data</span><span class="op">$</span><span class="va">MAD</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">MultiRD_results</span><span class="op">$</span><span class="va">ct.mad</span>,<span class="va">reffree_results</span><span class="op">$</span><span class="va">ct.mad</span>,<span class="va">SCDC_results</span><span class="op">$</span><span class="va">ct.mad</span><span class="op">)</span><span class="op">)</span>
<span class="va">data</span><span class="op">$</span><span class="va">CCC</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">MultiRD_results</span><span class="op">$</span><span class="va">ct.ccc</span>,<span class="va">reffree_results</span><span class="op">$</span><span class="va">ct.ccc</span>,<span class="va">SCDC_results</span><span class="op">$</span><span class="va">ct.ccc</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span> <span class="va">Cell_Types</span>, y<span class="op">=</span><span class="va">Methods</span>, size<span class="op">=</span><span class="va">MAD</span>, color<span class="op">=</span><span class="va">CCC</span>, group<span class="op">=</span> <span class="va">Cell_Types</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradient</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"yellow2"</span>,  high <span class="op">=</span> <span class="st">"blue"</span>, space <span class="op">=</span> <span class="st">"Lab"</span>, limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.001</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html" class="external-link">scale_size</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"WWR"</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">15</span><span class="op">)</span>,plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="MultiRD_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div id="real-bulk-rna-seq-data" class="section level2">
<h2 class="hasAnchor">
<a href="#real-bulk-rna-seq-data" class="anchor"></a>Real Bulk RNA-seq Data</h2>
<p>Now we move forward to analyze a bulk RNA-seq data generated from pancreatic islets by <a href="https://www.pnas.org/content/111/38/13924" class="external-link">Fadista et al. (2014)</a>. We focus on 77 samples with 51 considered as healthy (hbA1C level no larger than 6) and 26 considered as having type 2 diabetes (T2D; hbA1C larger than 6). To obtain the subject-level cell type proportions, we adopted SCDC-ENSAMBLE approach, a SCDC-based method aimed to implement deconvolution via integrating multiple single-cell reference sets (<a href="https://academic.oup.com/bib/article/22/1/416/5699815" class="external-link">Dong et al. (2021)</a>). scRNA-seq data from <a href="https://www.sciencedirect.com/science/article/pii/S2405471216302666?via%3Dihub" class="external-link">Baron et al. (2016)</a> and <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe et al. (2016)</a> are considered. To allow the potentially different gene expression patterns between the cases and controls, we implemented ENSEMBLE procedures for the samples from the two classes separately. (may take 5-10 minutes to run)</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fadista_77</span> <span class="op">&lt;-</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/fadista_77.rds"</span><span class="op">)</span>
<span class="co">#obtain subject-level proportions</span>
<span class="co">#for normal samples</span>
<span class="va">index_normal</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">hba1c_class2</span><span class="op">==</span><span class="st">"Normal"</span><span class="op">)</span>
<span class="va">fadista.healthy.ens</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/SCDC_ENSEMBLE.html" class="external-link">SCDC_ENSEMBLE</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="va">fadista_77</span><span class="op">[</span>,<span class="va">index_normal</span><span class="op">]</span>, sc.eset.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>baronh <span class="op">=</span> <span class="va">baron</span><span class="op">$</span><span class="va">sc.eset.qc</span>, segerh <span class="op">=</span> <span class="va">seger</span><span class="op">$</span><span class="va">sc.eset.qc</span><span class="op">)</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>, sample <span class="op">=</span> <span class="st">"sample"</span>, truep <span class="op">=</span> <span class="cn">NULL</span>, ct.sub <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>,<span class="st">"acinar"</span>,<span class="st">"ductal"</span><span class="op">)</span>, search.length <span class="op">=</span> <span class="fl">0.01</span>, grid.search <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">prop_est_normal</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/wt_prop.html" class="external-link">wt_prop</a></span><span class="op">(</span><span class="va">fadista.healthy.ens</span><span class="op">$</span><span class="va">w_table</span><span class="op">[</span><span class="fl">7</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, proplist <span class="op">=</span> <span class="va">fadista.healthy.ens</span><span class="op">$</span><span class="va">prop.only</span><span class="op">)</span>

<span class="co">#for diseased samples</span>
<span class="va">index_nnormal</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">hba1c_class2</span><span class="op">!=</span><span class="st">"Normal"</span><span class="op">)</span>
<span class="va">fadista.t2d.ens</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/SCDC_ENSEMBLE.html" class="external-link">SCDC_ENSEMBLE</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="va">fadista_77</span><span class="op">[</span>,<span class="va">index_nnormal</span><span class="op">]</span>, sc.eset.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>baronh <span class="op">=</span> <span class="va">baron</span><span class="op">$</span><span class="va">sc.eset.qc</span>, segerh <span class="op">=</span> <span class="va">seger</span><span class="op">$</span><span class="va">sc.eset.qc</span><span class="op">)</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>, sample <span class="op">=</span> <span class="st">"sample"</span>, truep <span class="op">=</span> <span class="cn">NULL</span>, ct.sub <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>,<span class="st">"acinar"</span>,<span class="st">"ductal"</span><span class="op">)</span>, search.length <span class="op">=</span> <span class="fl">0.01</span>, grid.search <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">prop_est_t2d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/wt_prop.html" class="external-link">wt_prop</a></span><span class="op">(</span><span class="va">fadista.t2d.ens</span><span class="op">$</span><span class="va">w_table</span><span class="op">[</span><span class="fl">7</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, proplist<span class="op">=</span><span class="va">fadista.t2d.ens</span><span class="op">$</span><span class="va">prop.only</span><span class="op">)</span>

<span class="co">#combine two groups </span>
<span class="va">prop_est</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">prop_est_normal</span>, <span class="va">prop_est_t2d</span><span class="op">)</span>
<span class="va">prop_est</span> <span class="op">=</span> <span class="va">prop_est</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">fadista_77</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prop_est</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">comb_sample</span> <span class="op">=</span> <span class="va">prop_est</span></code></pre></div>
<p>Next, to obtain the population-level mean proportions, we applied SCDC-ENSEMBLE again to the external bulk expression data from <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005694" class="external-link">Bunt et al. (2015)</a> at <a href="https://chencxxy28.github.io/MultiRD/articles/web/data.html">data download</a> and then averaged subject-level proportions for each cell type (Not run).</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Bunt</span> <span class="op">&lt;-</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/Bunt.rds"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">Bunt</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">&lt;-</span><span class="st">"sample"</span>
<span class="co">#obtain population-level mean proportions</span>
<span class="va">bunt.ens</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/SCDC_ENSEMBLE.html" class="external-link">SCDC_ENSEMBLE</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="va">Bunt</span>, sc.eset.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>baronh <span class="op">=</span> <span class="va">baron</span><span class="op">$</span><span class="va">sc.eset.qc</span>, segerh <span class="op">=</span> <span class="va">seger</span><span class="op">$</span><span class="va">sc.eset.qc</span><span class="op">)</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>, sample <span class="op">=</span> <span class="st">"sample"</span>, truep <span class="op">=</span> <span class="cn">NULL</span>, ct.sub <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>,<span class="st">"acinar"</span>,<span class="st">"ductal"</span><span class="op">)</span>, search.length <span class="op">=</span> <span class="fl">0.01</span>, grid.search <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">prop_est_bunt</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/wt_prop.html" class="external-link">wt_prop</a></span><span class="op">(</span><span class="va">bunt.ens</span><span class="op">$</span><span class="va">w_table</span><span class="op">[</span><span class="fl">7</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, proplist <span class="op">=</span> <span class="va">bunt.ens</span><span class="op">$</span><span class="va">prop.only</span><span class="op">)</span>
<span class="va">ave_est</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.subj.html">pop.ct.prop.subj</a></span><span class="op">(</span><span class="va">prop_est_bunt</span><span class="op">)</span></code></pre></div>
<p>The code above may take a long time to run. We have provided the estimated cell-type proportions, named <code>prop_est_bunt.rds</code> at <a href="https://chencxxy28.github.io/MultiRD/articles/web/data.html">data download</a>. We can directly use the estimates to calculate population-level mean proportions.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prop_est_bunt</span> <span class="op">=</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/prop_est_bunt.rds"</span><span class="op">)</span>
<span class="va">ave_est</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.subj.html">pop.ct.prop.subj</a></span><span class="op">(</span><span class="va">prop_est_bunt</span><span class="op">)</span></code></pre></div>
<p>Finally, given a list of pre-specified marker genes, named <code>fad_list_marker.rds</code> available at <a href="https://chencxxy28.github.io/MultiRD/articles/web/data.html">data download</a>, we applied MultiRD algorithm to conduct deconvolution. To allow the potentially different gene expression patterns between the cases and controls, we implemented ENSEMBLE procedures for the samples from the two classes separately (may take 5-10 minutes to run).</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read the list of marker genes</span>
<span class="va">list_marker</span> <span class="op">=</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">'https://github.com/chencxxy28/MultiRD/raw/master/vignettes/data/fad_list_marker.rds'</span><span class="op">)</span>

<span class="co">#normal samples</span>
<span class="va">MultiRD.object.normal</span> <span class="op">=</span> <span class="fu"><a href="../reference/MultiRD.onegroup.html">MultiRD.onegroup</a></span><span class="op">(</span>bulk.data<span class="op">=</span><span class="va">fadista_77</span><span class="op">[</span>,<span class="va">index_normal</span><span class="op">]</span>,list.marker<span class="op">=</span><span class="va">list_marker</span>,celltype.unique<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>,<span class="st">"acinar"</span>,<span class="st">"ductal"</span><span class="op">)</span>,subject.level.proportion<span class="op">=</span><span class="va">comb_sample</span><span class="op">[</span><span class="va">index_normal</span>,<span class="op">]</span>,population.level.proportion<span class="op">=</span><span class="va">ave_est</span>, tol<span class="op">=</span><span class="fl">0.0001</span><span class="op">)</span>
<span class="va">prop.est.normal.MultiRD</span> <span class="op">=</span> <span class="fu"><a href="../reference/MultiRD.predict.prop.html">MultiRD.predict.prop</a></span><span class="op">(</span>MultiRD.output<span class="op">=</span><span class="va">MultiRD.object.normal</span><span class="op">)</span>
<span class="va">prop.est.normal.reffree</span> <span class="op">=</span> <span class="fu"><a href="../reference/reffree.predict.prop.html">reffree.predict.prop</a></span><span class="op">(</span>MultiRD.output<span class="op">=</span><span class="va">MultiRD.object.normal</span><span class="op">)</span>

<span class="co">#diseased samples</span>
<span class="va">MultiRD.object.nnormal</span> <span class="op">=</span> <span class="fu"><a href="../reference/MultiRD.onegroup.html">MultiRD.onegroup</a></span><span class="op">(</span>bulk.data<span class="op">=</span><span class="va">fadista_77</span><span class="op">[</span>,<span class="va">index_nnormal</span><span class="op">]</span>,list.marker<span class="op">=</span><span class="va">list_marker</span>,celltype.unique<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>,<span class="st">"acinar"</span>,<span class="st">"ductal"</span><span class="op">)</span>,subject.level.proportion<span class="op">=</span><span class="va">comb_sample</span><span class="op">[</span><span class="va">index_nnormal</span>,<span class="op">]</span>,population.level.proportion<span class="op">=</span><span class="va">ave_est</span>, tol<span class="op">=</span><span class="fl">0.0001</span><span class="op">)</span>
<span class="va">prop.est.nnormal.MultiRD</span> <span class="op">=</span> <span class="fu"><a href="../reference/MultiRD.predict.prop.html">MultiRD.predict.prop</a></span><span class="op">(</span>MultiRD.output<span class="op">=</span><span class="va">MultiRD.object.nnormal</span><span class="op">)</span>
<span class="va">prop.est.nnormal.reffree</span> <span class="op">=</span> <span class="fu"><a href="../reference/reffree.predict.prop.html">reffree.predict.prop</a></span><span class="op">(</span>MultiRD.output<span class="op">=</span><span class="va">MultiRD.object.nnormal</span><span class="op">)</span>

<span class="co">#combine two groups for MultiRD</span>
<span class="va">MultiRD.est</span><span class="op">&lt;-</span><span class="va">comb_sample</span>
<span class="va">MultiRD.est</span><span class="op">[</span><span class="va">index_normal</span>,<span class="op">]</span><span class="op">&lt;-</span><span class="va">prop.est.normal.MultiRD</span>
<span class="va">MultiRD.est</span><span class="op">[</span><span class="va">index_nnormal</span>,<span class="op">]</span><span class="op">&lt;-</span><span class="va">prop.est.nnormal.MultiRD</span>

<span class="co">#combine two groups for reffree</span>
<span class="va">reffree.est</span><span class="op">&lt;-</span><span class="va">comb_sample</span>
<span class="va">reffree.est</span><span class="op">[</span><span class="va">index_normal</span>,<span class="op">]</span><span class="op">&lt;-</span><span class="va">prop.est.normal.reffree</span>
<span class="va">reffree.est</span><span class="op">[</span><span class="va">index_nnormal</span>,<span class="op">]</span><span class="op">&lt;-</span><span class="va">prop.est.nnormal.reffree</span></code></pre></div>
<p>To compare the results from MultiRD to other methods, we visualize the estimated proportions as follows:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MultiRD.est</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">MultiRD.est</span><span class="op">)</span>
<span class="va">MultiRD.est</span><span class="op">$</span><span class="va">id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MultiRD.est</span><span class="op">)</span>
<span class="va">reffree.est</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">reffree.est</span><span class="op">)</span>
<span class="va">reffree.est</span><span class="op">$</span><span class="va">id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">reffree.est</span><span class="op">)</span>
<span class="va">SCDC.est</span> <span class="op">=</span> <span class="va">comb_sample</span>
<span class="va">SCDC.est</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">SCDC.est</span><span class="op">)</span>
<span class="va">SCDC.est</span><span class="op">$</span><span class="va">id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">SCDC.est</span><span class="op">)</span>
<span class="va">Groups</span> <span class="op">=</span> <span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">hba1c_class2</span>
<span class="va">Groups</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">Groups</span><span class="op">)</span>
<span class="va">Groups</span><span class="op">$</span><span class="va">id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MultiRD.est</span><span class="op">)</span>

<span class="va">est_MultiRD</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">MultiRD.est</span>, id.vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"id"</span><span class="op">)</span><span class="op">)</span>
<span class="va">est_scdc</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">SCDC.est</span>, id.vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"id"</span><span class="op">)</span><span class="op">)</span>
<span class="va">est_reffree</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">reffree.est</span>, id.vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"id"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">est_MultiRD</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">est_scdc</span><span class="op">)</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">est_reffree</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"id"</span>, <span class="st">"Cell"</span>, <span class="st">"value"</span> <span class="op">)</span>
<span class="va">plot.data</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">est_MultiRD</span>,<span class="va">est_scdc</span>,<span class="va">est_reffree</span><span class="op">)</span>

<span class="va">plot.data</span><span class="op">$</span><span class="va">Methods</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"MultiRD"</span>,<span class="fl">462</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"SCDC"</span>,<span class="fl">462</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Ref-free"</span>,<span class="fl">462</span><span class="op">)</span><span class="op">)</span>
<span class="va">plot.data</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">plot.data</span>,<span class="va">Groups</span>,by<span class="op">=</span><span class="st">"id"</span><span class="op">)</span>
<span class="va">plot.data</span><span class="op">$</span><span class="va">order</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">plot.data</span><span class="op">$</span><span class="va">Groups</span><span class="op">==</span><span class="st">"Normal"</span>,<span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>
<span class="va">plot.data</span><span class="op">&lt;-</span><span class="va">plot.data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">plot.data</span><span class="op">$</span><span class="va">order</span><span class="op">)</span>,<span class="op">]</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">plot.data</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Methods</span>, y<span class="op">=</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Methods</span>, shape<span class="op">=</span><span class="va">Groups</span>, color<span class="op">=</span><span class="va">Groups</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span>, alpha <span class="op">=</span> <span class="fl">0.8</span>,position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">0.25</span>, height<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span> values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">'grey80'</span>,<span class="st">"Black"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_shape_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">21</span>,<span class="fl">24</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Cell</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Cell-type Proportions"</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">15</span><span class="op">)</span>,legend.position<span class="op">=</span><span class="st">"bottom"</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_fill_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Methods"</span>, guide <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="MultiRD_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<p>To evaluate the negative correlation between HbA1c levels and the beta cell functions, we constructed a generalized additive model using the estimated beta cell-type proportions as responses, some covariates (age, BMI, gender) as linear effects, and HbA1c as non-linear effect. The results from MultiRD are presented as follows.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">mgcv</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="va">age</span><span class="op">&lt;-</span><span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">[[</span><span class="st">"age"</span><span class="op">]</span><span class="op">]</span>
<span class="va">bmi</span><span class="op">&lt;-</span><span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">[[</span><span class="st">"bmi"</span><span class="op">]</span><span class="op">]</span>
<span class="va">gender</span><span class="op">&lt;-</span><span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">[[</span><span class="st">"gender"</span><span class="op">]</span><span class="op">]</span>
<span class="va">hba1c</span><span class="op">&lt;-</span><span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">[[</span><span class="st">"hba1c"</span><span class="op">]</span><span class="op">]</span>
<span class="va">fit_gam</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/mgcv/man/gam.html" class="external-link">gam</a></span><span class="op">(</span><span class="va">MultiRD.est</span><span class="op">[</span>,<span class="st">'beta'</span><span class="op">]</span><span class="op">~</span><span class="va">age</span><span class="op">+</span><span class="va">bmi</span><span class="op">+</span><span class="va">gender</span><span class="op">+</span><span class="fu"><a href="https://rdrr.io/pkg/mgcv/man/s.html" class="external-link">s</a></span><span class="op">(</span><span class="va">hba1c</span><span class="op">)</span>,family<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/mgcv/man/Beta.html" class="external-link">betar</a></span><span class="op">(</span>link<span class="op">=</span><span class="st">"logit"</span><span class="op">)</span><span class="op">)</span>
<span class="va">data_predict</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>age<span class="op">=</span><span class="fl">0</span>,bmi<span class="op">=</span><span class="fl">0</span>,gender<span class="op">=</span><span class="st">"Female"</span>,hba1c<span class="op">=</span><span class="va">hba1c</span><span class="op">)</span>

<span class="co">#plot</span>
<span class="va">fit_plot</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/mgcv/man/gam.html" class="external-link">gam</a></span><span class="op">(</span><span class="va">MultiRD.est</span><span class="op">[</span>,<span class="st">'beta'</span><span class="op">]</span><span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/mgcv/man/s.html" class="external-link">s</a></span><span class="op">(</span><span class="va">hba1c</span><span class="op">)</span>,family<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/mgcv/man/Beta.html" class="external-link">betar</a></span><span class="op">(</span>link<span class="op">=</span><span class="st">"logit"</span><span class="op">)</span><span class="op">)</span>
<span class="va">Groups</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">hba1c</span><span class="op">&lt;=</span><span class="fl">6</span>,<span class="st">"Normal"</span>,<span class="st">"T2D"</span><span class="op">)</span>
<span class="va">data_plot</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Cell_proportions<span class="op">=</span><span class="va">MultiRD.est</span><span class="op">[</span>,<span class="st">'beta'</span><span class="op">]</span>,hbaic<span class="op">=</span><span class="va">fadista_77</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span><span class="op">[[</span><span class="st">"hba1c"</span><span class="op">]</span><span class="op">]</span>,fitted<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/fitted.values.html" class="external-link">fitted</a></span><span class="op">(</span><span class="va">fit_plot</span><span class="op">)</span><span class="op">)</span>
<span class="va">nice_plot</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">data_plot</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">hbaic</span>, y <span class="op">=</span> <span class="va">Cell_proportions</span><span class="op">)</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.7</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">1.8</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="va">Groups</span>,color<span class="op">=</span><span class="va">Groups</span><span class="op">)</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">fitted</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1</span>, color<span class="op">=</span><span class="st">"darkgreen"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"HbA1c"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Beta Cell Proportions"</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">15</span><span class="op">)</span>,legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>
<span class="va">nice_plot</span></code></pre></div>
<p><img src="MultiRD_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chixiang Chen.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
